apps/api/src/domain/models/user-management/group/entities/invariant-validation/coscradUserGroupInvariantValidation.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { CoscradDataType, getCoscradDataSchema } from '@coscrad/data-types';
+import { InternalError } from '../../../../../../lib/errors/InternalError';
+import { DTO } from '../../../../../../types/DTO';
+import InvalidCoscradUserGroupDTOError from '../../../../../domainModelValidators/errors/InvalidCoscradUserGroupDTOError';
+import { Valid } from '../../../../../domainModelValidators/Valid';
+import { AggregateType } from '../../../../../types/AggregateType';
+import { Aggregate } from '../../../../aggregate.entity';
+import assertCoscradDataTypeError from '../../../../__tests__/invariant-validation-helpers/assertCoscradDataTypeError';
+import buildDummyUuid from '../../../../__tests__/utilities/buildDummyUuid';
+import { CoscradUserGroup } from '../coscrad-user-group.entity';
+
+/**
+ * TODO Let's extract the following to a separate utility- possibly
+ * even within the `@coscrad/data-types` lib.
+ */
+type ValueAndDescription<T = unknown> = {
+    value: T;
+    description: string;
+};
+
+type CoscradDataSchema = {
```

Comment:
Do we need this type defined locally? Why not use the one from `@coscrad/data-types`?

Reply:
The one in the lib is not nearly as strict.